sdk/azcore/arm/resource_identifier.go

```diff
@@ -0,0 +1,206 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package arm
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	providersKey             = "providers"
+	subscriptionsKey         = "subscriptions"
+	resourceGroupsLowerKey   = "resourcegroups"
+	locationsKey             = "locations"
+	builtInResourceNamespace = "Microsoft.Resources"
+)
+
+var (
+	// RootResourceID defines the tenant as the root parent of all other ResourceID.
+	RootResourceID = &ResourceID{
+		Parent:       nil,
+		ResourceType: TenantResourceType,
+		Name:         "",
+	}
+)
+
+// ResourceID represents a resource ID such as `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myRg`.
+// Don't create this type directly, use ParseResourceID instead.
+type ResourceID struct {
+	Parent            *ResourceID
```

Comment:
Can you please add doc comments to these fields as well?  Similar to what you did for `ResourceType`